src/flatpakutils.py

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/python3
+#
+# eos-flatpak-utils: Helper utilities to handle flatpak apps in EOS
+#
+# Copyright (C) 2017 Endless Mobile, Inc.
+# Authors:
+#  Mario Sanchez Prada <mario@endlessm.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import argparse
+import logging
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import gi
+gi.require_version('Flatpak', '1.0')
+from gi.repository import Flatpak
+from gi.repository import Gio
+from gi.repository import GLib
+gi.require_version('OSTree', '1.0')
+from gi.repository import OSTree
+
+from utils import die
+
+# Path to the flatpak system installation.
+FLATPAK_SYSTEM_INSTALLATION = '/var/lib/flatpak'
+
+
+def _look_up_ostree_ref_for_app_id(repo_path, app_id, remote_filter=None):
+    ostree_repo = OSTree.Repo.new(Gio.File.new_for_path(repo_path))
+    ostree_repo.open()
+
+    target_ref = None
+    _, refs = ostree_repo.list_refs()
+
+    logging.info("Searching for {} ref in {}".format(app_id, str(refs)))
+    for refspec in refs.keys():
+        _, remote, ref = OSTree.parse_refspec(refspec)
+
+        if remote_filter and remote != remote_filter:
+            logging.info("{} != {}, skipping...".format(remote, remote_filter))
+            continue
+
+        if str.startswith(ref, 'app/{}'.format(app_id)):
```

Comment:
Probably this is just fine, but one thing I'd suggest after having a second look is to make the test `app/{}/` with the trailing /. That make sure you only catch `com.bar.foo` and not `com.bar.football`.